designs/sam_publish_app_cmd.rst

```diff
@@ -0,0 +1,296 @@
+.. contents:: **Table of Contents**
+   :depth: 2
+   :local:
+
+``sam publish app`` command
+====================================
+
+This is the design for a command to publish an application to `AWS Serverless Application Repository (SAR)`_ with a SAM
+template. It can be used to create a new application w/ its first version, update existing application's metadata, and
```

Comment:
This could be something we could auto suggest, or implement flags for --bump-minor/patch

Reply:
We thought about doing this like [npm version](https://docs.npmjs.com/cli/version), but this is out of scope for publish currently.